catalog/spatial/registry/registry-api-impl/src/main/java/org/codice/ddf/registry/api/impl/RegistryStoreImpl.java

```diff
@@ -246,10 +246,7 @@ public SourceResponse query(QueryRequest request) throws UnsupportedQueryExcepti
 
         SourceResponse registryQueryResponse = super.query(request);
         for (Result singleResult : registryQueryResponse.getResults()) {
-            if (singleResult.getMetacard()
-                    .getAttribute(RegistryObjectMetacardType.REGISTRY_ID)
-                    .getValue()
-                    .toString()
+            if (RegistryUtility.getRegistryId(singleResult.getMetacard())
```

Comment:
getRegistryId returns a default of null and can throw null pointer here if the REGISTRY_ID is not found. Not likely but the hole is there.